### PR TITLE
Allow larger attributes by using new <ServerSideRender> prop

### PR DESCRIPTION
**Pull request by kienstra**
#### Changes
Even if there is a lot of text, the block preview will still render:

block-here

Fixes issues reported in BL: [#477](https://github.com/getblocklab/block-lab/issues/477), [#436](https://github.com/getblocklab/block-lab/issues/436), [#511](https://github.com/getblocklab/block-lab/issues/511)

#### Testing instructions
1. Please see 'Steps to reproduce' in [BHH-569](https://wpengine.atlassian.net/browse/BH-569?focusedCommentId=318658)
1. Please test with current version of WP and 5.4.2. The expected behavior is the same.

### CODE REVIEW

File: php/Blocks/Loader.php [hunk 1]
```diff
@@ -112,25 +99,10 @@ public function get_data( $key ) {
...
-	/**
-	 * Gets the callback for an action or filter.
-	 *
-	 * Enables keeping these methods protected,
-	 * while allowing actions and filters to call them.
-	 *
-	 * @param string $method_name The name of the method to get the callback for.
-	 * @return callable An enclosure that calls the function.
-	 */
-	protected function get_callback( $method_name ) {
```
Comment (kienstra) [inline php/Blocks/Loader.php:124]: Not related to this PR, but it's not necessary.

I wanted to keep methods `protected`, and use `->get_callback` to expose them in `add_action()` and `add_filter()`.

But this is verbose and unnecessary 🤣

**Comment by lukecarbis** [comment]
🎉

File: js/blocks/components/edit.js [hunk 2]
```diff
@@ -38,6 +38,7 @@ const Edit = ( { blockProps, block } ) => {
 						block={ `genesis-custom-blocks/${ block.name }` }
 						attributes={ attributes }
 						className="genesis-custom-blocks-editor__ssr"
+						httpMethod="POST"
```
Comment (kienstra) [inline js/blocks/components/edit.js:41]: A [new](https://github.com/WordPress/gutenberg/tree/master/packages/server-side-render#httpmethod) `prop` for this component.

**Comment by kienstra** [comment]
Hi @nickcernis and @mindctrl,
Could you please review this when you have a chance? 

This fixes an issue where the REST API endpoint for the `<ServerSideRender>` couldn't handle an `attributes` object over a certain size. So it uses the newly-added [httpMethod prop](https://github.com/WordPress/gutenberg/tree/master/packages/server-side-render#httpmethod) of that component to make it use `POST` requests, which allow a bigger `attributes` object.

[BHH-569](https://wpengine.atlassian.net/browse/BH-569?focusedCommentId=318658) has more details.

Thanks!

**Review by nickcernis — APPROVED** [review]
Working great for me with a million "a"s, @kienstra.  [About 1MB of text.]

Works under 5.4.2 too. Great job covering that.

[millionas.txt](https://github.com/studiopress/genesis-custom-blocks/files/5371490/millionas.txt)

File: php/Blocks/Loader.php [hunk 3]
```diff
@@ -610,4 +582,30 @@ public function add_field( $block_name, $field_config ) {
...
 		$this->blocks[ "genesis-custom-blocks/{$block_name}" ]['fields'][ $field_config['name'] ] = $field_config;
 	}
+
+	/**
+	 * Adds 'POST' to the allowed REST methods for GCB blocks.
+	 *
+	 * The <ServerSideRender> uses the httpMethod of 'POST' to handle a larger attributes object.
+	 * That is already added in WP 5.6+, so no need to add it there.
+	 *
+	 * @todo: Delete when this plugin's 'Requires at least' is bumped to 5.6.
```
Comment (nickcernis) [inline php/Blocks/Loader.php:592]: Good idea to add this reminder. Will be nice when we can bump the minimum version to 5.6.

**Review by mindctrl — APPROVED** [review]

File: php/Blocks/Loader.php (hunk 3 above)
Comment (kienstra) [reply php/Blocks/Loader.php:592]: Thanks! Yeah, that'll be good!

**Comment by kienstra** [comment]
Thanks so much for reviewing this.

**Pull request merged**